### ISSUE TITLE
Set comprehension is inferred as set[Generator[T, Any, Any]] instead of set[T]

### ISSUE THREAD

**Issue by RaptDept** [microsoft/pyright#164]
**Describe the bug**
Pyrite infers a set comprehension of expected type `Set[T]` as `Set[Generator[T, Any, Any]]`, producing errors when assigning this set to variables that are declared as `Set[T]`.

**To Reproduce**
See code.

**Expected behavior**
The inferred type of a set comprehension should instead be `Set[T]`, or at least suppress errors when assigning it as a `Set[T]`.

**Screenshots or Code**
```
from typing import Set

def func(s: Set[int]) -> None:
    pass

set_literal = {0, 1, 2} # set[int]
func(set_literal) # OK

set_comprehension = {i for i in range(3)} # set[Generator[int, Any, Any]]

set_comprehension_coerce: Set[int] = set_comprehension # error: Expression of type 'set[Generator[int, Any, Any]]' cannot be assigned to declared type 'Set[int]'

func(set_comprehension) # error: Argument of type 'set[Generator[int, Any, Any]]' cannot be assigned to parameter 's' of type 'Set[int]'
```

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool?
Yes.

Which version? You can find the version of the VS Code extension by clicking on the Pyright icon in the extensions panel.
1.0.43

**Additional context**

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next version of pyright.

**Comment by erictraut** [comment]
This is now fixed in 1.0.44, which I just published.
